RecoBTag/FeatureTools/plugins/DeepDoubleXTagInfoProducer.cc

```diff
@@ -34,322 +35,297 @@
-    , min_jet_pt_(iConfig.getParameter<double>("min_jet_pt"))
-    , min_candidate_pt_(iConfig.getParameter<double>("min_candidate_pt"))
-    , jet_token_(consumes<edm::View<reco::Jet>>(
-          iConfig.getParameter<edm::InputTag>("jets")))
-    , vtx_token_(consumes<VertexCollection>(
-          iConfig.getParameter<edm::InputTag>("vertices")))
-    , sv_token_(consumes<SVCollection>(
-          iConfig.getParameter<edm::InputTag>("secondary_vertices")))
-    , shallow_tag_info_token_(consumes<BoostedDoubleSVTagInfoCollection>(
-          iConfig.getParameter<edm::InputTag>("shallow_tag_infos")))
-{
-    produces<DeepDoubleXTagInfoCollection>();
-}
-
-DeepDoubleXTagInfoProducer::~DeepDoubleXTagInfoProducer()
-{
+DeepDoubleXTagInfoProducer::DeepDoubleXTagInfoProducer(const edm::ParameterSet& iConfig)
+    : jet_radius_(iConfig.getParameter<double>("jet_radius")),
+      min_jet_pt_(iConfig.getParameter<double>("min_jet_pt")),
+      min_candidate_pt_(iConfig.getParameter<double>("min_candidate_pt")),
+      jet_token_(consumes<edm::View<reco::Jet>>(iConfig.getParameter<edm::InputTag>("jets"))),
+      vtx_token_(consumes<VertexCollection>(iConfig.getParameter<edm::InputTag>("vertices"))),
+      sv_token_(consumes<SVCollection>(iConfig.getParameter<edm::InputTag>("secondary_vertices"))),
+      shallow_tag_info_token_(
+          consumes<BoostedDoubleSVTagInfoCollection>(iConfig.getParameter<edm::InputTag>("shallow_tag_infos"))) {
+  produces<DeepDoubleXTagInfoCollection>();
 }
 
-void DeepDoubleXTagInfoProducer::fillDescriptions(
-    edm::ConfigurationDescriptions& descriptions)
-{
-    // pfDeepDoubleXTagInfos
-    edm::ParameterSetDescription desc;
-    desc.add<edm::InputTag>("shallow_tag_infos",
-        edm::InputTag("pfBoostedDoubleSVAK8TagInfos"));
-    desc.add<double>("jet_radius", 0.8);
-    desc.add<double>("min_jet_pt", 150);
-    desc.add<double>("min_candidate_pt", 0.95);
-    desc.add<edm::InputTag>("vertices", edm::InputTag("offlinePrimaryVertices"));
-    desc.add<edm::InputTag>("secondary_vertices",
-        edm::InputTag("inclusiveCandidateSecondaryVertices"));
-    desc.add<edm::InputTag>("jets", edm::InputTag("ak8PFJetsCHS"));
-    descriptions.add("pfDeepDoubleXTagInfos", desc);
+DeepDoubleXTagInfoProducer::~DeepDoubleXTagInfoProducer() {}
+
+void DeepDoubleXTagInfoProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // pfDeepDoubleXTagInfos
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("shallow_tag_infos", edm::InputTag("pfBoostedDoubleSVAK8TagInfos"));
+  desc.add<double>("jet_radius", 0.8);
+  desc.add<double>("min_jet_pt", 150);
+  desc.add<double>("min_candidate_pt", 0.95);
+  desc.add<edm::InputTag>("vertices", edm::InputTag("offlinePrimaryVertices"));
+  desc.add<edm::InputTag>("secondary_vertices", edm::InputTag("inclusiveCandidateSecondaryVertices"));
+  desc.add<edm::InputTag>("jets", edm::InputTag("ak8PFJetsPuppi"));
```

Comment:
It is hard to verify which updates are coming from this PR, only meant to introduce DeepDoubleX V2, because you added also all recent clang-tidy checks related updates.

While backporting those updates is annoying for the review, but still acceptable for the code, I notice that you are also backporting this line change, which comes from #28936, and it is certainly not due here.
Please fix it!

And, if I can suggest, I would also only backport the changes for DeepDoubleX V2, letting aside the non-necessary code-checks related cleanings (well, if those cleanings are just a few, and simple enough, there is nothing wrong in backporting also them: but not in files as long and complicated as this one)

Reply:
Sorry, must have done something wrong when rebasing. I'll fix it up.

Reply:
For some reason git went nuts, trying to apply changes to this file. Anyway, should be cleaner now.